src/frequenz/channels/_anycast.py

```diff
@@ -151,6 +151,16 @@ async def send(self, msg: T) -> None:
             self._chan.recv_cv.notify(1)
 
 
+class _Sentinel:
+    """A sentinel value to indicate that a value has not been set."""
+
+    def __repr__(self) -> str:
+        return "_Sentinel"
+
+    def __bool__(self) -> bool:
+        return False
```

Comment:
Nitpicking here, but why are `__repr__` and `__bool__` needed here? Also I would call it `_Empty`, `_Absent` or something like that. It makes more obvious what is it for and `assert self._next is not _Empty` reads more easily than `assert self._next is not _Sentinel`.

Reply:
But the problem with `bool` is that you first should make sure it is an `_Empty` instance, otherwise it could be a user sending a `False`, `None`, or anything that also supports `__bool__` as a valid message, so I'm not convinced is a good idea to even support it.

Reply:
`if not vv:` allows users to check for all False cases together,  for example for `list[int] | None | _Empty`.  Else they'll have to do: `if vv is not _Empty and vv is not None and len(vv) == 0:`

If they just want to check if it is empty,  they'd do `if vv is _Empty`,  just like `if vv is None`, etc if they just want to do those checks.

Reply:
done.